webauthn-server-core/src/main/java/com/yubico/webauthn/OriginMatcher.java

```diff
@@ -16,38 +16,56 @@ static boolean isAllowed(
         boolean allowPort,
         boolean allowSubdomain
     ) {
+        log.trace("isAllowed({}, {}, {}, {})", origin, allowedOrigins, allowPort, allowSubdomain);
+
+        final URL originUrl;
+        {
+            URL originUrl1;
+            try {
+                originUrl1 = new URL(origin);
+            } catch (MalformedURLException e) {
+                log.debug("Origin in client data is an invalid URL; will only match exactly: {}", origin);
+                originUrl1 = null;
+            }
+            originUrl = originUrl1;
+        }
+
         return allowedOrigins.stream().anyMatch(allowedOriginString -> {
             if (allowedOriginString.equals(origin)) {
+                log.debug("Exact match: {} == {}", origin, allowedOriginString);
                 return true;
-            } else {
+            } else if (originUrl != null && (allowPort || allowSubdomain)) {
                 final URL allowedOrigin;
                 try {
                     allowedOrigin = new URL(allowedOriginString);
                 } catch (MalformedURLException e) {
-                    log.error("Allowed origin is an invalid URL: {}", allowedOriginString);
+                    log.error("Allowed origin is an invalid URL; skipped for port/subdomain matching: {}", allowedOriginString);
```

Comment:
I would like to change "skipped for port...etc" to "skipping port...etc". "skipped for" *could* imply that the URL is being skipped for now so that it will be tested for port/subdomain matching later. Of course, this interpretation does not make sense in this context, but if you do not fully understand the context and all you have access to is log messages, it is possible that "skipped for" could give you the wrong impression.

Reply:
Good points, I agree. Thanks!